src/test/java/org/openrewrite/java/migrate/lang/SwitchCaseAssigningToSwitchExpressionTest.java

```diff
@@ -0,0 +1,657 @@
+                          default: System.out.println("foo");
+                      }
+                  }
+              }
+              """
+          ));
+    }
+
+    @Test
+    void inlineWhenVariableOnlyToBeReturned() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  String doFormat() {
+                      String formatted;
+                      switch (1) {
+                          default: formatted = "foo"; break;
+                      }
+                      return formatted;
+                  }
+              }
+              """,
+            """
+              class Test {
+                  String doFormat() {
+                      return switch (1) {
+                          default: yield "foo";
+                      };
+                  }
+              }
+              """
+          ));
+    }
+
+    @Test
+    void doNotinlineWhenInappropriate() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  String originalVariableNotReturned() {
+                      String formatted;
+                      switch (1) {
+                          default: formatted = "foo"; break;
+                      }
+                      return "string";
+                  }
+
+                  String codeBetweenSwitchAndReturn() {
+                      String formatted;
+                      switch (1) {
+                          default: formatted = "foo"; break;
+                      }
+                      System.out.println("Hey");
+                      return formatted;
+                  }
+
+                  void noReturnedExpression() {
+                      String formatted;
+                      switch (1) {
+                          default: formatted = "foo"; break;
+                      }
+                      return;
+                  }
+              }
+              """,
+            """
+              class Test {
+                  String originalVariableNotReturned() {
+                      String formatted= switch (1) {
+                          default: yield "foo";
+                      };
+                      return "string";
+                  }
+
+                  String codeBetweenSwitchAndReturn() {
+                      String formatted= switch (1) {
+                          default: yield "foo";
+                      };
+                      System.out.println("Hey");
+                      return formatted;
+                  }
+
+                  void noReturnedExpression() {
+                      String formatted= switch (1) {
+                          default: yield "foo";
+                      };
+                      return;
+                  }
+              }
+              """
+          ));
+    }
+
+    @Test
+    @ExpectedToFail
+    void failsToFormatWithASpaceWhenOriginalVariableHasNoInitializer() {
```

Comment:
@timtebeek Would appreciate a pointer on how to fix that formatting bug, this is the result when `formatted` doesn't have an initializer:
```
String formatted= switch (1) {
```